docs/proposals/image-provenance.md

```diff
@@ -0,0 +1,320 @@
+(whether due to user not having permission due to bad image, or some other permission reason)
+and throttling itself and surfacing the error in a way that CLIs and UIs can show to the user.
+
+## Changes in policy over time
+
+The Backend might change the policy over time.  For example, yesterday `redis:v1` was allowed, but today `redis:v1` is not allowed
+due to a CVE that just came out (fictional scenario).  In this scenario:
+. 
+
+- a newly created replicaSet will be unable to create Pods.
+- updating a deployment will be safe in the sense that it will detect that the new ReplicaSet is not scaling
+  up and not scale down the old one.
+- an existing replicaSet will not be unable to create Pods that replace ones which are terminated.  If this is due to
+  slow loss of nodes, then there should be time to react before significant loss of capacity.
+- For non-replicated things (size 1 ReplicaSet, PetSet), a single node failure may disable it.
+- a node rolling update will eventually check for liveness of replacements, and would be throttled if
+  in the case when the image was no longer allowed and so replacements could not be started.
+- rapid node restarts will cause existing pod objects to be restared by kubelet.
+- slow node restarts or network partitions will cause node controller to delete pods and there will be no replacement
+
+It is up to the Backend implementor, and the cluster administrator who decides to use that backend, to decide
+whether the Backend should be allowed to change its mind.  There is a tradeoff between responsiveness
+to changes in policy, versus keeping existing services running.  The two models that make sense are:
+
+- never change a policy, unless some external process has ensured no active objects depend on the to-be-forbidden
+  images.
+- change a policy and assume that transition to new image happens faster than the existing pods decay.
+
+## Ubernetes
+
+If two clusters share an image policy backend, then they will have the same policies.
+
+The clusters can pass different tokens to the backend, and the backend can use this to distinguish
+between different clusters.
+
+## Image tags and IDs
+
+Image tags are like: `myrepo/myimage:v1`.
+
+Image IDs are like: `myrepo/myimage@sha256:beb6bd6a68f114c1dc2ea4b28db81bdf91de202a9014972bec5e4d9171d90ed`.
+You can see image IDs with `docker images --no-trunc`.
+
+The Backend needs to be able to resolve tags to IDs (by talking to the images repo).
+If the Backend resolves tags to IDs, there is some risk that the tag-to-ID mapping will be
+modified after approval by the Backend, but before Kubelet pulls the image.  We will not address this
+race condition at this time.
+
+We will wait and see how much demand there is for closing this hole. If the community demands a solution,
+we may suggest one of these:
+
+1.  Use a backend that refuses to accept images that are specified with tags, and require users to resolve to IDs
```

Comment:
I just commented on this in https://github.com/kubernetes/kubernetes/pull/27129/files/4a60be831efce93d7e210df47d79e7c18d5d13c2#r71607506

I think it makes sense to either map tag to SHA in kubectl, in CICD, or after the fact in Kubelet.  I agree the admission controller is a bad place to map tag to SHA.  I think admission is a fine place to require image names that use SHAs.